hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/SingleTransactionBlockItems.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2023-2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.state;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.Timestamp;
+import com.hedera.hapi.node.base.TokenType;
+import com.hedera.hapi.node.base.Transaction;
+import com.hedera.node.app.spi.workflows.record.SingleTransaction;
+import com.swirlds.common.exceptions.NotImplementedException;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/**
+ * A record of a single transaction, including the transaction, record and the sidecar records.
+ */
+public record SingleTransactionBlockItems(
+        @NonNull Transaction transaction,
+//        @NonNull TransactionResult transactionResult,
+        @Nullable TransactionOutputs transactionOutputs
+//        @NonNull StateChanges stateChanges,
+        ) implements SingleTransaction {
+    public SingleTransactionBlockItems {
+        requireNonNull(transaction, "transaction must not be null");
+//        requireNonNull(transactionResult, "transactionResult must not be null");
+        requireNonNull(transactionOutputs, "transactionOutputs must not be null");
+//        requireNonNull(stateChanges, "stateChanges must not be null");
+    }
+
+    // This is used by BlockStream, and is not serialized.
+    public record TransactionOutputs(@Nullable TokenType tokenType) {}
```

Comment:
Is this not a protobuf message ?

Reply:
No it's not. I'm not aware of any protobuf that groups these block items. There's `Block`, of course, but that's for a round not a single transaction. Did I miss something that has these already?